logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -17,21 +17,23 @@ import com.wire.kalium.network.api.message.MessageApi
 import com.wire.kalium.network.api.message.MessagePriority
 import com.wire.kalium.network.exceptions.QualifiedSendMessageError
 import com.wire.kalium.network.utils.isSuccessful
+import com.wire.kalium.persistence.dao.ConversationDAO
 import com.wire.kalium.persistence.dao.message.MessageDAO
 import com.wire.kalium.persistence.dao.message.MessageEntity
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.firstOrNull
 import kotlinx.coroutines.flow.map
 
 interface MessageRepository {
-    suspend fun getMessagesForConversation(conversationId: ConversationId, limit: Int): Flow<List<Message>>
-    suspend fun persistMessage(message: Message): Either<CoreFailure, Unit>
+    suspend fun getMessagesForConversation(conversationId: ConversationId, limit: Int, offset: Int): Flow<List<Message>>
+    suspend fun persistMessage(message: Message, isMyMessage: Boolean = true): Either<CoreFailure, Unit>
```

Comment:
What is `myMessage`?

Is it when I am the sender? Could it be from myself but from another device?

Can't we just figure it out internally based on `userId` and `clientId`?